### PR TITLE
feat: image url

### PR DESCRIPTION
**Pull request by JohannesMessner**
Goals:

This adds the type `ImageUrl`.

For now it can:
- load the uri to tensor
- load the uri to bytes

TODO:
- [x] more tests
- [x] appropriate validation
- [x] some small TODOs in the code comments

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -11,6 +11,8 @@ pydantic = "^1.10.2"
 numpy = "^1.23.4"
 protobuf = { version = "^4.21.9", optional = true }
 torch = { version = "^1.0.0", optional = true }
+pillow = {version = "^9.3.0", extras = ["image"]}
```
Comment (samsja) [inline pyproject.toml:14]: we neeed to mark them optional I believe

File: pyproject.toml [hunk 2]
```diff
@@ -11,6 +11,8 @@ pydantic = "^1.10.2"
...
+types-pillow = {version = "^9.3.0.1", extras = ["image"]}
```
Comment (samsja) [inline pyproject.toml:15]: same here

File: docarray/typing/url/helper.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import urllib.request
+from contextlib import nullcontext
+
+
+def _uri_to_blob(uri: str, timeout=None) -> bytes:
+    """Convert uri to blob
+    Internally it reads uri into blob.
+    :param uri: the uri of Document
+    :param timeout: timeout for urlopen. Only relevant if uri is not local
+    :return: blob bytes.
```
Comment (samsja) [inline docarray/typing/url/helper.py:12]: Why this helper file ? I think this should all go to the `any_url` file

**Review by samsja — COMMENTED** [review]
one comment

**Review by dongxiang123 — APPROVED** [review]

**Review by samsja — APPROVED** [review]

**Pull request merged**